### PR TITLE
Fixes #3830 and #3655. (IPC bugfix collection)

### LINKED ISSUE #3830: Trying to implant stuff into IPCs does not work
The surgery type of implanting stuff, with drills. It just gives this

You start making some space inside Nymphaea's  cavity with the surgical drill.

and then you can't put ANYTHING inside that cavity, and it makes that same response for everything

### PR DESCRIPTION
**Pull request by MarinaGryphon**
Open to suggestions for other IPC-related issues to fix in this PR.

### CODE REVIEW

**Comment by Lohikar** [comment]
This PR does not compile.

File: code/modules/surgery/robotics.dm [hunk 1]
```diff
@@ -197,7 +197,7 @@
 		"<span class='warning'>You cause a short circuit in [target]'s [affected.name]!</span>")
 		target.apply_damage(rand(5,10), BURN, affected)
 
-/datum/surgery_step/robotics/fix_organ_robotic //For artificial organs
+/datum/surgery_step/robotics/fix_ORGAN_ROBOT //For artificial organs
```
Comment (Lohikar) [inline code/modules/surgery/robotics.dm]: Why did you rename these? Typepaths should not have capitals.

File: code/modules/surgery/robotics.dm [hunk 2]
```diff
@@ -265,7 +265,7 @@
 			if(I)
 				I.take_damage(rand(3,5),0)
 
-/datum/surgery_step/robotics/detatch_organ_robotic
+/datum/surgery_step/robotics/detatch_ORGAN_ROBOT
```
Comment (Lohikar) [inline code/modules/surgery/robotics.dm]: Same here.

File: code/modules/surgery/robotics.dm [hunk 3]
```diff
@@ -315,7 +315,7 @@
 		user.visible_message("<span class='warning'>[user]'s hand slips, disconnecting \the [tool].</span>", \
 		"<span class='warning'>Your hand slips, disconnecting \the [tool].</span>")
 
-/datum/surgery_step/robotics/attach_organ_robotic
+/datum/surgery_step/robotics/attach_ORGAN_ROBOT
```
Comment (Lohikar) [inline code/modules/surgery/robotics.dm]: And here.

**Comment by Lohikar** [comment]
Still doesn't compile, by the way.

File: code/modules/surgery/robotics.dm (hunk 1 above)
Comment (MarinaGryphon) [reply code/modules/surgery/robotics.dm]: Ahahaha oops this was uh. A find-and-replace to fix the thing that made it not compile.

File: code/modules/surgery/robotics.dm [hunk 4]
```diff
@@ -6,6 +6,9 @@
 /datum/surgery_step/robotics/
 	can_infect = 0
 	can_use(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool)
+		if (isipc(target) && isipc(user) && user == target)
```
Comment (skull132) [inline code/modules/surgery/robotics.dm]: `if (user == target && isipc(user))` just collapses this better.

File: code/modules/surgery/robotics.dm [hunk 5]
```diff
@@ -428,3 +431,146 @@
...
+		..()
+		var/obj/item/organ/external/chest/affected = target.get_organ(target_zone)
+		if (affected.implants.len)
+			var/fail_prob = 10
+			fail_prob += 100 - tool_quality(tool)
+			if (prob(fail_prob))
+				var/obj/item/weapon/implant/imp = affected.implants[1]
+				user.visible_message("<span class='warning'>Something beeps inside [target]'s [affected.name]!</span>")
+				playsound(imp.loc, 'sound/items/countdown.ogg', 75, 1, -3)
+				spawn(25)
```
Comment (skull132) [inline code/modules/surgery/robotics.dm]: Timer!

**Comment by skull132** [comment]
Why is a separate surgery even necessary? What's the issue with the original one?

File: code/modules/surgery/robotics.dm (hunk 5 above)
Comment (MarinaGryphon) [reply code/modules/surgery/robotics.dm]: This was copied from the implant file, so a timer would need to be used there, too. What's the proper syntax for it?

**Comment by MarinaGryphon** [comment]
For one, it supposedly wasn't working. Two, Lohikar said it probably needed its own surgery, as the one for organics doesn't make sense for IPCs.

**Comment by Lohikar** [comment]
Status on this?

**Comment by MarinaGryphon** [comment]
Not tested yet, please don't merge. Changes have been made, though.

File: code/modules/surgery/robotics.dm (hunk 5 above)
Comment (MarinaGryphon) [reply code/modules/surgery/robotics.dm]: Resolved.

File: code/modules/surgery/robotics.dm (hunk 4 above)
Comment (MarinaGryphon) [reply code/modules/surgery/robotics.dm]: Fixed.

**Comment by Lohikar** [comment]
It's been almost 20 days. Tested yet?

**Comment by Lohikar** [comment]
Stale & no response from author. Comment if you pick this up again.
